### PR TITLE
Add guard on Python single goal action server example

### PR DESCRIPTION
**Pull request by mrjogo**
Fixes https://github.com/ros2/examples/issues/371

### CODE REVIEW

File: rclpy/actions/minimal_action_server/examples_rclpy_minimal_action_server/server_single_goal.py [hunk 1]
```diff
@@ -100,15 +100,20 @@ def execute_callback(self, goal_handle):
...
-        result.sequence = feedback_msg.sequence
+            # Populate result message
+            result = Fibonacci.Result()
+            result.sequence = feedback_msg.sequence
 
-        self.get_logger().info('Returning result: {0}'.format(result.sequence))
+            self.get_logger().info('Returning result: {0}'.format(result.sequence))
 
-        return result
+            return result
```
Comment (fujitatomoya) [inline rclpy/actions/minimal_action_server/examples_rclpy_minimal_action_server/server_single_goal.py:116]: why these lines needs to be under `_goal_lock` lock? i think goal_handle has been succeeded, so we can release the lock after that, then return the result.

Comment (mrjogo) [reply rclpy/actions/minimal_action_server/examples_rclpy_minimal_action_server/server_single_goal.py]: Good point. Fixed.

**Review by fujitatomoya — APPROVED** [review]
lgtm with green CI

**Comment by mrjogo** [comment]
> @mrjogo can you fix DCO error? i think you just missed `--signoff` for the 2nd commit.

Sorry, gets me every time. Done.

**Review by ahcorde — APPROVED** [review]

**Comment by ahcorde** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/21086/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/15389/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/21803/)

**Comment by ahcorde** [comment]
https://github.com/Mergifyio backport jazzy

**Comment by ahcorde** [comment]
@fujitatomoya Do we need to backport this to `iron` or `humble`?

**Comment by fujitatomoya** [comment]
@ahcorde would be nice. even though this is an example code, users likely copy&paste the examples, i think.

**Comment by ahcorde** [comment]
https://github.com/Mergifyio backport humble iron

**Pull request merged**